### PR TITLE
change Speed and Distance to be interfaces as store distances and speed in their native units....

### PR DESCRIPTION
**Pull request by lp117937**
... to avoid reducing accuracy through converting back and forth

### CODE REVIEW

File: distance.go [hunk 1]
```diff
@@ -7,77 +7,46 @@ package units
...
-// Internally, it stores the distance in metres, as a float64.
-// Create a new Distance by multiplying one of the constants:
-// d1 := 100 * units.Metre
-// d2 := 10 * units.NauticalMile
-// Use methods like Metres() or Kilometres() to get the distance in the unit of your choice.
-// US and UK spelling of consts and methods supported.
-
-type Distance float64
+// Distance represents a quantity of speed.
+type Distance interface {
```
Comment (rscstar) [inline distance.go:17]: need 
Short() string
Name() string
in the interface for units.Distance.Short() to work.
see test "TestDistanceShort"

Actually all of the functions on a unit should be in the interface that way they can be called like this
fmt.Println(d.Short(), d.Name(), d.String())
Valid()
String()

File: distance.go [hunk 2]
```diff
@@ -7,77 +7,46 @@ package units
...
+	Unit
```
Comment (rscstar) [inline distance.go:18]: I assume this is a comment?

File: distance_test.go [hunk 3]
```diff
@@ -7,49 +7,58 @@ package units_test
...
+	"math/rand"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/starboard-nz/units"
 )
 
-const ε = 0.000000001
+const δ = 0.000001
```
Comment (rscstar) [inline distance_test.go:20]: Not sure about this. I don't know how to reproduce that special character. 
Would it be better to use basic ascii "delta =" or "cDelta"

File: speed.go [hunk 4]
```diff
@@ -7,55 +7,34 @@ package units
...
-
-// Mpss return the speed in m/s.
-func (s Speed) Mpss() float64 {
-	return float64(s)
-}
+type (
+	Knot float64
+	Kph  float64
+	Mps  float64
+	Mph  float64
```
Comment (rscstar) [inline speed.go]: Maybe comment 
Mps // meters per second
Mph // miles per hour

File: distance.go (hunk 1 above)
Comment (lp117937) [reply distance.go:17]: Ah... it's in `unit.go`, which I forgot to add, added now.

File: distance.go (hunk 2 above)
Comment (lp117937) [reply distance.go:18]: No, it adds the `Unit` interface, defined in `unit.go`, which I forgot to `git add`. It's there now

File: distance_test.go (hunk 3 above)
Comment (lp117937) [reply distance_test.go:20]: you can always copy & paste :smile:

File: speed.go (hunk 4 above)
Comment (lp117937) [reply speed.go]: added comments

**Review by rscstar — APPROVED** [review]
Looks good. This will be nice to use in lots of places.

**Pull request merged**